genie-core/src/main/java/com/netflix/genie/core/jpa/services/JpaBaseService.java

```diff
@@ -44,12 +49,19 @@
 @Getter(AccessLevel.PACKAGE)
 class JpaBaseService {
 
+    static final ObjectMapper MAPPER;
     static final String GENIE_TAG_NAMESPACE = "genie.";
     static final String GENIE_ID_TAG_NAMESPACE = GENIE_TAG_NAMESPACE + "id:";
     static final String GENIE_NAME_TAG_NAMESPACE = GENIE_TAG_NAMESPACE + "name:";
     private static final char COMMA = ',';
     private static final String EMPTY_STRING = "";
 
+    static {
+        final DateFormat iso8601 = new GenieDateFormat();
+        iso8601.setTimeZone(TimeZone.getTimeZone("UTC"));
+        MAPPER = new ObjectMapper().registerModule(new Jdk8Module()).setDateFormat(iso8601);
```

Comment:
Does this mean that if someone sends a JSON with a string field containing a date in a format other than ISO8601 when we serialize then deserialize, the string value for that field will change?
This might trip some users, if they rely on a specific format for setting and parsing back dates from the metadata they store.
Although that is probably the case anyway, wether you explicitly use ISO8601 or not.

Reply:
This is consistently how we serialize dates from the database and is mentioned in the documentation. I'm not sure there's much else we can do there.